readme.md

```diff
@@ -0,0 +1,183 @@
+# Socket sharding in Linux with Go
+
+How many times that we were working on the terminal with multiple tabs and we launched an HTTP server, and then we forgot that the server was already being executed, and we tried to relaunch it from another tab getting the known error:
+
+```bash
+go run main.go 
+listen tcp :8080: bind: address already in use
+```
+This is because we cannot open a socket with the same source address and port by default in Linux and the vast majority of operating systems.
+
+### Socket options
+
+When we create a new TCP socket on a Linux, we can set options that affect the behaviour of the socket. For example, one of these options is SO_REUSEPORT, which allows multiple sockets to bind to the same IP address and port. With this feature, the Linux kernel distributes incoming requests across all the sockets that share the same address and port combination, getting a load balancing inside the Kernel.
+
+> SO_REUSEPORT
+> 
+> For TCP sockets, this option allows accept(2) load distribution in a multi-threaded server to be improved by using a distinct listener socket for each thread.  This provides improved load distribution as compared to traditional techniques such using a single accept(2)ing thread that distributes connections, or having multiple threads that compete to accept(2) from the same socket.
+> 
+> For UDP sockets, the use of this option can provide better distribution of incoming datagrams to multiple processes (or threads) as compared to the traditional technique of having multiple processes compete to receive datagrams on the same socket.
+> 
+> https://man7.org/linux/man-pages/man7/socket.7.html
+
+![](https://user-images.githubusercontent.com/8400576/127754771-6a789f00-7022-4602-a628-a3d124b43421.png)
+
+As we can notice, we not only obtain the super power to create more than one socket with the same IP: Port combination, we also obtain a kind of load balancer in the kernel mode.
```

Comment:
"Obtain" se refiere a algo como que perdiste y que re-encontraste. Mejor...

```suggestion
As we can notice, we not only get the super power to create more than one socket with the same IP: Port combination, but we also obtain a kind of load balancer in the kernel mode.
```

También hay una modificación menor al team de que si comenzaste con "we not only get" (que es una comparación) sino que necesitas tener un "pero" (but) en la siguiente idea.